### PR TITLE
feat(ProposalScreen): screen, animations, fixes home-governance

### PR DESCRIPTION
**Pull request by gHashTag**
New components, logic, screens

## Bug

I ran into a gap when creating a method for sending a deposit transaction

#### `this.sendMsg` output:

```js
{"tx_response": {"code": 2, "codespace": "sdk", "data": "", "events": [], "gas_used": "0", "gas_wanted": "0", "height": "0", "info": "", "logs": [], "raw_log": "math/big: cannot unmarshal \"500\\u00a0000\\u00a0000\\u00a0000\\u00a0000\\u00a0000\" into a *big.Int: tx parse error", "timestamp": "", "tx": null, "txhash": "D6D3F4296D288F040A63E2F91E0D289DE84C57AD9B5831BD1337669FAEEF65F7"}}
```

Even if you pass a number that is in the bigInt range, the error remains the same

I transmit all the data correctly, that is, the real address, the existing proposal id, the processed amount, I did by analogy with the `delegate` method

image

### Update 1

If you send 20 tokens, the following error appears

```
{"tx_response": {"code": 4, "codespace": "sdk", "data": "", "events": [], "gas_used": "0", "gas_wanted": "0", "height": "0", "info": "", "logs": [], "raw_log": "signature verification failed; please verify account number (11) and chain-id (haqq_121799-1): feePayer pubkey EthPubKeySecp256k1{024270DBEFEB9EF29445FF223642C84EA746E706DDC61D0292F795CE585C835332} is different from transaction pubkey EthPubKeySecp256k1{026DB2833451C5AEFC8AF10B05787D07F456057CD7ED33416B5D1D0E6AB81A624A}: invalid pubkey [evmos/ethermint@v0.19.3/app/ante/eip712.go:253]: unauthorized", "timestamp": "", "tx": null, "txhash": "B38C4CF1EA0AC37AE7B21E7EBA1C2FFD7BF6F7FDAA62AB9571A23546AC44F5FF"}}
```

### CODE REVIEW

File: src/components/home-governance/home-governance.tsx [hunk 1]
```diff
@@ -1,88 +1,102 @@
...
+export const HomeGovernance = ({onPressCard}: HomeGovernanceProps) => {
   const cosmos = useCosmos();
   const {proposals, setStatusFilter, statusFilter} = useProposals();
+  const [refreshing, setRefreshing] = useState(false);
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    cosmos.syncGovernanceVoting();
-  });
+    cosmos.syncGovernanceVoting().finally(() => setLoading(false));
```
Comment (vivalaakam) [inline src/components/home-governance/home-governance.tsx:29]: move on screen

File: src/components/home-governance/home-governance.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  loading?: boolean;
+}
+
+export const HomeGovernance = ({
+  onPressCard,
+  onRefresh,
+  loading,
+  refreshing,
+}: HomeGovernanceProps) => {
+  const {proposals, setStatusFilter, statusFilter} = useProposals();
```
Comment (vivalaakam) [inline src/components/home-governance/home-governance.tsx:27]: move it ons screen

Comment (vivalaakam) [reply src/components/home-governance/home-governance.tsx:27]: Or you can move on screen on proposal fetching data

File: src/components/home-governance/voting-card/voting-card-active.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+const colorVotes = {
+  yes: Color.graphicGreen1,
+  no: Color.textRed1,
+  abstain: Color.graphicSecond4,
+  veto: Color.textYellow1,
+};
+
+export const VotingCardActive = ({id, onPress}: VotingCardActiveProps) => {
+  const item = useMemo(() => {
```
Comment (vivalaakam) [inline src/components/home-governance/voting-card/voting-card-active.tsx:45]: move on screen

File: src/components/home-governance/voting-card/voting-card-completed.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+type VotingCardCompletedProps = {
+  id: number;
+
+  onPress?: (id: number) => void;
+};
+
+export const VotingCardCompleted = ({
+  id,
+  onPress,
+}: VotingCardCompletedProps) => {
```
Comment (vivalaakam) [inline src/components/home-governance/voting-card/voting-card-completed.tsx:30]: move on screen

File: src/components/home-governance/voting-card/voting-card-active.tsx (hunk 3 above)
Comment (vivalaakam) [inline src/components/home-governance/voting-card/voting-card-active.tsx:45]: use normal data from item of flat list

File: src/components/home-governance/voting-card/voting-card-completed.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  const item = useMemo(() => {
```
Comment (vivalaakam) [inline src/components/home-governance/voting-card/voting-card-completed.tsx:31]: use normal data from item of flat list

File: src/components/home-governance/voting-card/voting-card-completed.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  const isRejected = status === VotingCompletedStatuses.rejected;
+  const statusColor = isRejected ? Color.textRed1 : Color.textGreen1;
+
+  const handlePress = () => onPress?.(id);
+
+  return (
+    <Pressable onPress={handlePress} style={styles.backgroundContainer}>
+      <View style={styles.topInfoBlock}>
+        <Icon i18 color={statusColor} name={isRejected ? 'close' : 'check'} />
+        <Spacer width={5.5} />
```
Comment (vivalaakam) [inline src/components/home-governance/voting-card/voting-card-completed.tsx:49]: use decimal value

File: src/components/proposal/proposal.tsx [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+  const cardRef = useRef<VotingCardDetailRefInterface>();
+  const voteSelectedRef = useRef<VoteNamesType>();
+  const [vote, setVote] = useState<VoteNamesType>();
+  const [collectedDeposit, setCollectedDeposit] = useState(0);
+
+  const app = useApp();
+  const {visible} = useWalletsList();
+  const closeDistance = useWindowDimensions().height / 6;
+
+  const item = useMemo(() => {
```
Comment (vivalaakam) [inline src/components/proposal/proposal.tsx:42]: move on screen

File: src/components/proposal/proposal.tsx [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+  const onDeposit = () => {
+    showModal('wallets-bottom-sheet', {
+      wallets: visible,
+      closeDistance,
+      title: I18N.proposalAccountTitle,
+      eventSuffix: '-proposal-deposit',
+    });
+    const onDepositSubmit = async (address: string) => {
+      await item?.sendDeposit(address, 0.00000005);
```
Comment (vivalaakam) [inline src/components/proposal/proposal.tsx:54]: move on screen

File: src/components/ui/svg-icon.tsx [hunk 9]
```diff
@@ -298,6 +298,46 @@ export const NoTransactionsIcon = (props: SvgProps) => (
   </Svg>
 );
 
+export const NoProposalVotingIcon = (props: SvgProps) => (
```
Comment (vivalaakam) [inline src/components/ui/svg-icon.tsx:301]: why not regular icon?

File: src/components/ui/voting-line.tsx [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+export interface VotingLineInterface {
+  updateValues: (newVotes: VotesType) => void;
+  setSelected: React.Dispatch<React.SetStateAction<VoteNamesType>>;
+}
+
+const animConfig: WithTimingConfig = {
+  duration: 500,
+};
+
+export const VotingLine = memo(
```
Comment (vivalaakam) [inline src/components/ui/voting-line.tsx:34]: this is too specific component for UI

File: src/models/governance-voting.ts [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+        (acc: number, item: Coin) => acc + item.amount,
+        0,
+      );
+    } else {
+      return;
+    }
+  }
+
+  async sendDeposit(address: string, amount: number) {
+    const cosmos = new Cosmos(app.provider!);
```
Comment (vivalaakam) [inline src/models/governance-voting.ts:171]: IMHO: no need to mix the logic of database models and the logic of obtaining this very data, there are services for this

**Review by vivalaakam — CHANGES_REQUESTED** [review]
fix comments and conflicts

File: src/components/home-governance/voting-card/voting-card-active.tsx (hunk 3 above)
Comment (gHashTag) [reply src/components/home-governance/voting-card/voting-card-active.tsx:45]: If I use data from flatlist, I will have to manually write the transfer of all the necessary props 2 times

I also noticed a feature of classes that you can't throw the whole object with all the properties

File: src/models/governance-voting.ts (hunk 11 above)
Comment (gHashTag) [reply src/models/governance-voting.ts:171]: This is done in order not to write an extra parameter to the function and call the method directly from the element without using an extra render

image

File: src/screens/home-governance.tsx [hunk 12]
```diff
@@ -1,7 +1,45 @@
...
+
+  const {proposals, setStatusFilter, statusFilter} = useProposals();
+
+  const onSelect = (tag: ProposalsTagType) => () => {
+    setStatusFilter(tag[0]);
+  };
+
+  return (
+    <HomeGovernance
+      {...{
```
Comment (vivalaakam) [inline src/screens/home-governance.tsx:34]: looks like something wrong

File: src/services/cosmos.ts [hunk 13]
```diff
@@ -150,6 +153,49 @@ export class Cosmos {
...
   }
 
+  async deposit(source: string, proposalId: number, amount: number) {
+    try {
+      const sender = await this.getSender(source);
+      const memo = '';
+      const params = {
+        proposalId,
+        deposit: {
+          amount: ((amount ?? 0) * WEI).toLocaleString().replace(/,/g, ''),
```
Comment (vivalaakam) [inline src/services/cosmos.ts:163]: Why are you converting a number here to a local?

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**